Refactor and cleanup codebase

hamza1311 opened the pull request:
#### Description

<!-- Please include a summary of the change. -->

Fixes (part of) #1841 
Fixes https://github.com/yewstack/yew/issues/885 (by removing the service altogether)

#### Checklist

<!-- For further details, please read CONTRIBUTING.md -->

- [x] I have run `cargo make pr-flow`
- [ ] I have reviewed my own code
- [ ] I have added tests
  <!-- If this is a bug fix, these tests will fail if the bug is present (to stop it from cropping up again) -->
  <!-- If this is a feature, my tests prove that the feature works -->

hamza1311 commented:
This PR is ready for review.

I have left out moving `vdom` into it's own crate out of this PR because it is pretty big as is and that would just make it bigger. When that happens, `yew-validation` will become a part of that crate. `yew-router` rewrite (#1791) and `yew-agent` rewrite will also help in the cleanup process.

mc1098 reviewed (commented):
The function component macro tests seem to have been deleted and not moved to yew_macro.

The following example's READMEs have references to either yewtil or services:
- General README for examples refers to the 'reader service' for the `file_upload` description
- boids
- crm
- file_upload
- futures
- game_of_life
- router
- store
- timer
- todomvc

The Elements page under the Listeners heading has a code snippet that uses `ConsoleService` [Elements.md L135](https://github.com/hamza1311/yew/blob/refactor/website/docs/concepts/html/elements.md#L135)

It would be nice to add a section about `callback_future` on the [Callbacks doc page.](https://github.com/hamza1311/yew/blob/refactor/website/docs/concepts/components/callbacks.md)

Is it worth removing the warning on the [Function Components Introduction page](https://github.com/hamza1311/yew/blob/refactor/website/docs/concepts/function-components.md) as function components are being moved into the `yew` crate?

hamza1311 commented:
> The following example's READMEs have references to either yewtil or services:

> The Elements page under the Listeners heading has a code snippet that uses `ConsoleService` [Elements.md L135](https://github.com/hamza1311/yew/blob/refactor/website/docs/concepts/html/elements.md#L135)

Fixed those.

> It would be nice to add a section about `callback_future` on the [Callbacks doc page.](https://github.com/hamza1311/yew/blob/refactor/website/docs/concepts/components/callbacks.md)

I think that can come later, this PR is already *huge* and mainly focuses on refactor.

> Is it worth removing the warning on the [Function Components Introduction page](https://github.com/hamza1311/yew/blob/refactor/website/docs/concepts/function-components.md) as function components are being moved into the `yew` crate?

Yes, removed it.

hamza1311 commented:
@mc1098 feel free to take a look again, I made the changes.

mc1098 commented on examples/boids/README.md:
```diff
@@ -17,17 +17,15 @@ trunk serve --release
 ## Concepts
 
-The example uses [`IntervalService`] to drive the game loop.
+The example uses `gloo::timers` implementation of `setInterval` to drive the Yew game loop.
```
Nit:
```suggestion
The example uses [`gloo::timers`](https://gloo-rs.web.app/docs/timer) implementation of `setInterval` to drive the Yew game loop.
```

mc1098 commented on examples/file_upload/README.md:
```diff
@@ -7,7 +7,7 @@ The contents of the selected file are then rendered to the page either as a whol
 ## Concepts
 
-Demonstrates the use of `ReaderService`.
+Demonstrates reading from files in Yew with the help of `gloo::file`.
```
Nit:
```suggestion
Demonstrates reading from files in Yew with the help of [`gloo::file`](https://gloo-rs.web.app/docs/file).
```

mc1098 reviewed (commented):
Couple of links to add to `gloo` references but that's it :D

mc1098 reviewed (approved):
Looks good to me :D

mokeyish commented:
Hello everyone, what's the repleacement of https://yew.rs/concepts/services/fetch ？

hamza1311 commented:
> Hello everyone, what's the repleacement of https://yew.rs/concepts/services/fetch ？

#1841 mentions them.
